### PR TITLE
issue #7216

### PR DESCRIPTION
**Pull request by rachana-webkul**
## Issue Reference
#7216

## Description
Now on adding configurable product the minimum quantity of the configurable product is showing on the user side

### CODE REVIEW

**Comment by Sanjaybhattwebkul** [comment]
@jitendra-webkul 
As we discuss. i am putting this PR on hold

**Comment by Sanjaybhattwebkul** [comment]
@rachana-webkul 
**Not working**, please find the screenshot [Here](https://prnt.sc/z66B93JS6i09)

**Comment by Sanjaybhattwebkul** [comment]
@rachana-webkul 
Any update?

**Comment by Sanjaybhattwebkul** [comment]
@rachana-webkul 
The space should be according to the card content on the home page.

**Comment by Sanjaybhattwebkul** [comment]
@rachana-webkul 
Please resolve the **conflicts**

**Comment by Sanjaybhattwebkul** [comment]
@rachana-webkul 
**Not working**, The configurable product's **price** should be visible like: **$10.00**  _As low as_

**Comment by rachana-webkul** [comment]
@Sanjaybhattwebkul  now price-lable add

**Comment by rachana-webkul** [comment]
@Sanjaybhattwebkul 
Conflicts **resolved**, please verify

File: packages/Webkul/Product/src/Type/Configurable.php [hunk 1]
```diff
@@ -502,18 +502,71 @@ public function canBeMovedFromWishlistToCart($item)
     }
 
     /**
-     * Get product prices.
-     *
-     * @return array
-     */
+     * Return variants of configurable product.
```
Comment (devansh-webkul) [inline packages/Webkul/Product/src/Type/Configurable.php:505]: Add return type in heredoc.

File: packages/Webkul/Product/src/Type/Configurable.php [hunk 2]
```diff
@@ -502,18 +502,71 @@ public function canBeMovedFromWishlistToCart($item)
...
+    */
+    public function getAllowedProducts()
+    {
+        $product =  $this->product;
```
Comment (devansh-webkul) [inline packages/Webkul/Product/src/Type/Configurable.php:509]: No need of variable directly pass the value.

File: packages/Webkul/Product/src/Type/Configurable.php [hunk 3]
```diff
@@ -502,18 +502,71 @@ public function canBeMovedFromWishlistToCart($item)
...
+
+        return $variants;
+    }
+
+    /**
+     * Return minimum price of variants.
+    */
+    protected function getVariantPrice()
+    {
+        $product =  $this->product;
```
Comment (devansh-webkul) [inline packages/Webkul/Product/src/Type/Configurable.php:542]: Same here.

**Review by devansh-webkul — CHANGES_REQUESTED** [review]

**Comment by rachana-webkul** [comment]
@devansh-webkul 
resolved issue, please verify.

**Comment by Sanjaybhattwebkul** [comment]
@rachana-webkul 
**Not working**, please find the **screenshot** [Here](https://webkul.chatwhizz.com/share/screenshot/63ee098693ea2803a944bfae)

**Comment by rachana-webkul** [comment]
@Sanjaybhattwebkul 
product label added, please verify

File: packages/Webkul/Product/src/Type/Configurable.php [hunk 4]
```diff
@@ -502,18 +502,71 @@ public function canBeMovedFromWishlistToCart($item)
...
-     * Get product prices.
-     *
+     * Return variants of configurable product.
      * @return array
-     */
+     * 
+    */
+    public function getAllowedProducts()
+    {
+
```
Comment (devansh-webkul) [inline packages/Webkul/Product/src/Type/Configurable.php:511]: Remove one line from here.

**Review by devansh-webkul — CHANGES_REQUESTED** [review]

**Comment by rachana-webkul** [comment]
@devansh-webkul 
spaces resolved ,please verify

**Comment by rachana-webkul** [comment]
@Sanjaybhattwebkul 
configuration price fixed,please verify

File: packages/Webkul/Product/src/Type/Configurable.php [hunk 5]
```diff
@@ -502,18 +502,70 @@ public function canBeMovedFromWishlistToCart($item)
...
+                $variants[] = $variant;
+            }
+        }
+
+        return $variants;
+    }
+
+    /**
+     * Return minimum price of variants.
+     * @return array
```
Comment (devansh-webkul) [inline packages/Webkul/Product/src/Type/Configurable.php:539]: Add a new line and change the return type to float.

File: packages/Webkul/Product/src/Type/Configurable.php [hunk 6]
```diff
@@ -502,18 +502,70 @@ public function canBeMovedFromWishlistToCart($item)
...
+    */
+    protected function getVariantPrice()
+    {
+        $prices = [];
+
+        foreach ($this->getAllowedProducts($this->product) as $variant) {
+            $prices[$variant->id] = $variant->getTypeInstance()->getProductPrices();
```
Comment (devansh-webkul) [inline packages/Webkul/Product/src/Type/Configurable.php:546]: No need to add $prices[$variant->id] as your end result is just taking minimum from the array.

Just pass it like this,

~~~php
$prices[] = $variant->getTypeInstance()->getProductPrices();
~~~

**Review by devansh-webkul — CHANGES_REQUESTED** [review]

**Comment by rachana-webkul** [comment]
@devansh-webkul 
changes done,please verify

**Comment by Sanjaybhattwebkul** [comment]
@rachana-webkul 
**Not fixed,** the product is wrong on the home page,

**Comment by rachana-webkul** [comment]
@Sanjaybhattwebkul 
conflict resolved, please verify

**Comment by rachana-webkul** [comment]
@Sanjaybhattwebkul 
configurable variant price changed,please verify

**Comment by rachana-webkul** [comment]
@Sanjaybhattwebkul 
product price fixed for configurable products,please verify

**Comment by rachana-webkul** [comment]
@Sanjaybhattwebkul 
lowest variant price of configurable product is visible, please verify

**Comment by rachana-webkul** [comment]
@Sanjaybhattwebkul  conflicts resolved,please verify

**Comment by Sanjaybhattwebkul** [comment]
@rachana-webkul 
**Still** getting the **same issue**. please find the screenshot [Here](https://webkul.chatwhizz.com/share/screenshot/642a8535f0168c07d9e98ecd)

**Comment by rachana-webkul** [comment]
@Sanjaybhattwebkul  it's working please verify

**Review by Sanjaybhattwebkul — APPROVED** [review]

**Comment by Sanjaybhattwebkul** [comment]
@rachana-webkul 
This is **working fine.**

**Comment by Amitk-webkul** [comment]
@rachana-webkul 
with the tax on the Product page tax message layout is not set.[Here](https://prnt.sc/1TNG8JStrPcl)

**Comment by devansh-webkul** [comment]
> @rachana-webkul with the tax on the Product page tax message layout is not set.[Here](https://prnt.sc/1TNG8JStrPcl) After deleting a variant product. It is not adding products to the cart. [Here](https://tinyurl.com/292ucjll)

Is this reproducible in master, if so then we will fix in the new issue as this pull request scope is to get the min price from all variants.

**Comment by Amitk-webkul** [comment]
@devansh-webkul 
The above issues are not replicated in the master.

**Pull request closed without merging**